### PR TITLE
[PDI-17012] Issues with Carte Jobs API jobImage endpoint in Pentaho R…

### PR DESCRIPTION
**Pull request by LeonardoCoelho71950**
…EST API

@pentaho/tatooine 
@pentaho-lmartins @ssamora

### CODE REVIEW

File: engine/src/main/java/org/pentaho/di/www/JobMap.java [hunk 1]
```diff
@@ -128,6 +129,21 @@ public synchronized CarteObjectEntry getFirstCarteObjectEntry( String jobName )
     return null;
   }
 
+  public synchronized CarteObjectEntry getUniqueCarteObjectEntry( String jobName ) throws DuplicateKeyException {
```
Comment (pentaho-lmartins) [inline engine/src/main/java/org/pentaho/di/www/JobMap.java]: Can I get a few unit tests for this method?
1. `jobName` with null value
2. `jobName` with empty value
3. When only one job with the given name exists
4. When multiple jobs with the given name exist

File: engine/src/main/java/org/pentaho/di/www/GetJobImageServlet.java [hunk 2]
```diff
@@ -136,14 +139,20 @@ public void doGet( HttpServletRequest request, HttpServletResponse response ) th
     String jobName = request.getParameter( "name" );
     String id = request.getParameter( "id" );
 
+    boolean useXML = "Y".equalsIgnoreCase( request.getParameter( "xml" ) );
+
     // ID is optional...
     //
     Job job;
-    CarteObjectEntry entry;
+    CarteObjectEntry entry = null;
```
Comment (pentaho-lmartins) [inline engine/src/main/java/org/pentaho/di/www/GetJobImageServlet.java]: This is not necessary, the object will be initialized as `null`.

File: engine/src/main/java/org/pentaho/di/www/GetJobImageServlet.java [hunk 3]
```diff
@@ -185,6 +194,27 @@ public void doGet( HttpServletRequest request, HttpServletResponse response ) th
     }
   }
 
+  private void buildDuplicateResponse( HttpServletResponse response, String jobName, boolean useXML ) {
```
Comment (pentaho-lmartins) [inline engine/src/main/java/org/pentaho/di/www/GetJobImageServlet.java]: Please write a unit test for the method.

File: engine/src/main/java/org/pentaho/di/www/GetJobImageServlet.java [hunk 4]
```diff
@@ -136,14 +139,20 @@ public void doGet( HttpServletRequest request, HttpServletResponse response ) th
...
     // ID is optional...
     //
     Job job;
-    CarteObjectEntry entry;
+    CarteObjectEntry entry = null;
     if ( Utils.isEmpty( id ) ) {
-      // get the first transformation that matches...
-      //
-      entry = getJobMap().getFirstCarteObjectEntry( jobName );
+
```
Comment (pentaho-lmartins) [inline engine/src/main/java/org/pentaho/di/www/GetJobImageServlet.java]: Can I see a unit test to exercise the new logic?

**Comment by LeonardoCoelho71950** [comment]
Working on the unit tests

File: engine/src/main/java/org/pentaho/di/www/GetJobImageServlet.java (hunk 2 above)
Comment (LeonardoCoelho71950) [reply engine/src/main/java/org/pentaho/di/www/GetJobImageServlet.java]: Done

File: engine/src/main/java/org/pentaho/di/www/GetJobImageServlet.java (hunk 4 above)
Comment (LeonardoCoelho71950) [reply engine/src/main/java/org/pentaho/di/www/GetJobImageServlet.java]: Made several unit tests for GetJobImageServlet

File: engine/src/main/java/org/pentaho/di/www/GetJobImageServlet.java (hunk 3 above)
Comment (LeonardoCoelho71950) [reply engine/src/main/java/org/pentaho/di/www/GetJobImageServlet.java]: Made several unit tests for GetJobImageServlet

**Pull request merged**